os_migrate/plugins/module_utils/resource.py

```diff
@@ -529,3 +530,20 @@ def _validation_params_errors(self):
             if param not in params:
                 errors.append('Missing params.{0}.'.format(param))
         return errors
+
+    # Can be overridden in some subclasses.
+    def _validation_empty_name_errors(self):
+        """For resources which have a name property validate that it is non empty
+
+        Returns: list of strings (error messages)
+        """
+        errors = []
+        params = self.params()
+        name_param_key = 'name'
+        # check to see if this resource has a name
+        if name_param_key in self.params_from_sdk:
+            # Check to see if name is present and if so is it empty.
+            # _validation_params_errors check will catch if it is not present.
+            if name_param_key in params.keys() and not params.get(name_param_key, ''):
```

Comment:
The condition could simply be `if params.get(name_param_key, ''):`. Not a blocker but may be worth amending?

Reply:
This rather: `if not params.get(name_param_key):'

Reply:
True but then "params.name is empty." will be appended to errors even when "name" is missing. I figured that this is not expected for this validation as "name" missing is handled by _validation_params_errors.

Reply:
Actually now i realize why you did the `name_param_key in params.keys()` part -- to make sure we don't get duplicate validation errors for the same issue, right?  All good then, we can merge. Just will give some time for Carlos to have a chance to review as well.

Reply:
@jistr exactly :)